### DISCUSSION TITLE
Attempting to execute a .sql file using psycopg3

### DISCUSSION THREAD

**Discussion by Nikhil-Khetani** [psycopg/psycopg#467] [Q&A] [answered]
For my project, I would like to keep my SQL queries separate from any python modules, ie in a separate directory of .sql files. However, I wish to be able to execute them on my databases by calling a function in python. As I understand one could do this in previous versions of psycopg by reading the SQL file and passing it as a string. However, this is no longer possible due to the LiteralString typechecking. Does anyone have any ideas on how to get around this? Thanks.

**Comment by dlax** [comment, marked as answer]
For this use case, a `# type: ignore` seems  reasonable to me because it clearly indicates that the "source" is trusted, and it's *your* responsibility after all. I'm not sure it's worth adding some API (to build a `SQL` value from a plain `str`) on psycopg side, though.

**Reply by dvarrazzo** [reply]
Or, analogously, you can use a cast, and document that the operation is safe. More details in [the LiteralString pep](https://peps.python.org/pep-0675/#appendix-b-limitations).

**Reply by Nikhil-Khetani** [reply]
Thanks Denis! This does the job for me - I completely missed it before.

**Reply by Nikhil-Khetani** [reply]
> Or, analogously, you can use a cast, and document that the operation is safe. More details in [the LiteralString pep](https://peps.python.org/pep-0675/#appendix-b-limitations).

Thanks, this was really useful Daniele
